test/server/listener_manager_impl_test.cc

```diff
@@ -1899,6 +1946,142 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, OriginalDstTestFilterOptionFail)
   EXPECT_EQ(0U, manager_->listeners().size());
 }
 
+class OriginalDstTestFilterIPv6
+    : public Extensions::ListenerFilters::OriginalDst::OriginalDstFilter {
+  Network::Address::InstanceConstSharedPtr getOriginalDst(int) override {
+    return Network::Address::InstanceConstSharedPtr{
+        new Network::Address::Ipv6Instance("::0002", 2345)};
```

Comment:
We're going to have to condition this test to avoid running in IPv4 environments. See how this is done for other IPv6 tests, e.g. https://github.com/envoyproxy/envoy/blob/master/test/common/network/connection_impl_test.cc#L1275